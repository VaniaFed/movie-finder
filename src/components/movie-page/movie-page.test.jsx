import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { MoviePage } from './index';

describe('<MoviePage />', () => {
    const movie = {
        budget: 44000000,
        genres: ['Crime', 'Drama', 'Mystery', 'Western'],
        id: 273248,
        overview:
            'Bounty hunters seek shelter from a raging blizzard and get caught up in a plot of betrayal and deception.',
        poster_path:
            'https://image.tmdb.org/t/p/w500/fqe8JxDNO8B8QfOGTdjh6sPCdSC.jpg',
        release_date: '2015-12-25',
        revenue: 155760117,
        runtime: 167,
        tagline: 'No one comes up here without a damn good reason.',
        title: 'The Hateful Eight',
        vote_average: 7.6,
        vote_count: 5584
    };
    const tree = renderer
        .create(
            <MemoryRouter>
                <MoviePage movie={movie} />
            </MemoryRouter>
        )
        .toJSON();
    it('should be rendered with movie', () => {
        expect(tree).toMatchSnapshot();
    });
});
