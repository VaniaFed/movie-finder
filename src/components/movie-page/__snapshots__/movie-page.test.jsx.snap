// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviePage /> should be rendered with movie 1`] = `
Array [
  <div
    className="movie-block"
  >
    <div
      className="movie-header"
    >
      <h1
        className="logo"
      >
        netfixroulette
      </h1>
      <a
        href="/search"
        onClick={[Function]}
      >
        <button
          className="movie-header__search-button movie-header__search-button--reversed button"
          data-button="search-page__button-search"
          type="button"
        >
          search
        </button>
      </a>
    </div>
    <div
      className="movie movie-block__movie"
    >
      <img
        alt="movieimage"
        className="movie__image"
        src="https://image.tmdb.org/t/p/w500/fqe8JxDNO8B8QfOGTdjh6sPCdSC.jpg"
      />
      <div
        className="movie__info"
      >
        <h2
          className="movie__title"
          data-text="movie-page__movie-title"
        >
          The Hateful Eight
        </h2>
        <p
          className="movie__genres"
        >
          Crime, Drama, Mystery, Western
        </p>
        <p
          className="movie__date"
        >
          2015-12-25
        </p>
        <p
          className="movie__duration"
        >
          167
          m
        </p>
        <p
          className="movie__description"
        >
          Bounty hunters seek shelter from a raging blizzard and get caught up in a plot of betrayal and deception.
        </p>
      </div>
    </div>
  </div>,
  <div
    className="movie-page-info"
  >
    <span
      className="movie-page-info__text"
    >
      Films by
    </span>
    <span
      className="movie-page-info__genre"
    >
      Crime, Drama, Mystery, Western
    </span>
  </div>,
]
`;
